data/blog/mobile-app-launch-profiling.md

```diff
@@ -0,0 +1,46 @@
+---
+title: 'Mobile App Launch Profiling'
+date: '2024-04-17'
+tags: ['mobile', 'ios', 'android', 'profiling']
+draft: false
+summary: "See what's happening in your app before your first line of code can even run."
+images: ['/images/mobile-app-launch-profiling/launch-profile.png']
+layout: PostLayout
+canonicalUrl: mobile-app-launch-profiling
+authors: ['andrewmcknight']
+---
+
+If you use Sentry’s Performance monitoring on Cocoa and Android, you also have the ability to see [profiles](https://docs.sentry.io/product/profiling/) of how your app runs in the wild on real end user devices. Any time an automatic or manual transaction is recorded, that profiling data is attached. But are there times where you can’t record a transaction? 🤔
+
+# App launch performance
+
+<aside>
+💡 This post will focus on our Cocoa app launch profiling feature. You can read more about the Android counterpart by heading over to that area of our docs: https://docs.sentry.io/platforms/android/profiling/#app-start-profiling
+</aside>
+
+Apple places special focus on good [performance of an app’s launch](https://developer.apple.com/documentation/xcode/reducing-your-app-s-launch-time). They even built an entire process to enforce good performance: the [watchdog](https://developer.apple.com/documentation/xcode/addressing-watchdog-terminations). There are two ways you can monitor launch performance today:
+
+1. Instruments. You can profile your app while launching it on any devices you have available.
+2. MetricKit. You can see a histogram of app launch times in Xcode’s Organizer. This data is inexact, because you can’t correlate any specific context to problematic launches, and the way launch time is measured is out of your control:
+
+> The launch-time metric measures the time from the user tapping the app icon on their Home screen to the app drawing its first frame to the screen. If your app still has to run code after it has drawn its first frame, but before the user can begin using the app, that time doesn’t contribute to the launch-time metric. **Extra startup activities still contribute to the user’s perception of the app’s responsiveness**.
+
+So, if you or one of your dependencies does some work at launch time, you don’t necessarily see that in the Xcode Organizer.
+
+# Sentry app launch profiling
+
+Now, you can get profiling data from the moment your app starts executing code, before calls to `main` or `appDidFinishLaunching`, where the Sentry SDK is typically initialized and was previously the earliest moment a performance transaction could be recorded.
```

Comment:
```suggestion
Now, once you enable App Launch Profiling, you can get profiling data from the moment your app starts executing code. From before calls to `main` or `appDidFinishLaunching`, which is where the Sentry SDK is typically initialized and was previously the earliest moment a performance transaction could be recorded.
```

Reply:
This just makes it wordier without adding any additional information. The second sentence also becomes incomplete.